packages/nodes-base/nodes/Slack/SlackTrigger.node.ts

```diff
@@ -0,0 +1,419 @@
+					this,
+					'members',
+					'GET',
+					'/users.list',
+				)) as Array<{ id: string; name: string }>;
+				for (const user of users) {
+					returnData.push({
+						name: user.name,
+						value: user.id,
+					});
+				}
+
+				returnData.sort((a, b) => {
+					if (a.name < b.name) {
+						return -1;
+					}
+					if (a.name > b.name) {
+						return 1;
+					}
+					return 0;
+				});
+
+				return returnData;
+			},
+		},
+	};
+
+	webhookMethods = {
+		default: {
+			async checkExists(this: IHookFunctions): Promise<boolean> {
+				return true;
+			},
+			async create(this: IHookFunctions): Promise<boolean> {
+				return true;
+			},
+			async delete(this: IHookFunctions): Promise<boolean> {
+				return true;
+			},
+		},
+	};
+
+	async webhook(this: IWebhookFunctions): Promise<IWebhookResponseData> {
+		const filters = this.getNodeParameter('trigger', []) as string[];
+		const req = this.getRequestObject();
+		const options = this.getNodeParameter('options', {}) as IDataObject;
+		const binaryData: IBinaryKeyData = {};
+		const watchWorkspace = this.getNodeParameter('watchWorkspace', false) as boolean;
+
+		// Check if the request is a challenge request
+		if (req.body.type === 'url_verification') {
+			const res = this.getResponseObject();
+			res.status(200).json({ challenge: req.body.challenge }).end();
+
+			return {
+				noWebhookResponse: true,
+			};
+		}
+
+		// Check if the event type is in the filters
+		const eventType = req.body.event.type as string;
+
+		if (
+			!filters.includes('file_share') &&
+			!filters.includes('any_event') &&
+			!filters.includes(eventType)
+		) {
+			return {};
+		}
+
+		const eventChannel = req.body.event.channel ?? req.body.event.item.channel;
+
+		// Check for single channel
+		if (!watchWorkspace) {
+			if (
+				eventChannel !== (this.getNodeParameter('channelId', {}, { extractValue: true }) as string)
+			) {
+				return {};
+			}
+		}
+
+		// Check if user should be ignored
+		if (options.userIds) {
+			const userIds = options.userIds as string[];
+			if (userIds.includes(req.body.event.user)) {
+				return {};
+			}
+		}
+
+		if (options.resolveIds) {
+			if (req.body.event.user) {
+				if (req.body.event.type === 'reaction_added') {
+					req.body.event.user_resolved = await getUserInfo.call(this, req.body.event.user);
+					req.body.event.item_user_resolved = await getUserInfo.call(
+						this,
+						req.body.event.item_user,
+					);
+				} else {
+					req.body.event.user_resolved = await getUserInfo.call(this, req.body.event.user);
+				}
+			}
+
+			if (eventChannel) {
+				const channel = await getChannelInfo.call(this, eventChannel);
+				const channelResolved = channel;
+				req.body.event.channel_resolved = channelResolved;
+			}
+		}
+
+		let responseData: IDataObject = {};
+
+		if (
+			req.body.event.subtype === 'file_share' &&
+			(filters.includes('file_share') || filters.includes('any_event'))
+		) {
+			responseData = req.body.event.files;
```

Comment:
When I post a message with some file attachments, I can only see the files in the event and not the message itself. Should this be just `req.body.event`?